lib/main.js

```diff
@@ -7,10 +7,46 @@ const Reader = require('asset-pipe-js-reader');
 const body = require('body/json');
 const boom = require('boom');
 const uuid = require('uuid/v4');
+const { extname } = require('path');
+const { Transform } = require('stream');
 
 const params = require('./params');
 const schemas = require('./schemas');
 
+class CheckEmptyPayload extends Transform {
+    constructor() {
+        super();
+        this.chunks = [];
+    }
+
+    _transform(chunk, enc, cb) {
+        this.chunks.push(chunk);
+        cb();
+    }
+
+    _flush(cb) {
+        const buffered = this.chunks.map(chunk => chunk.toString()).join('');
+        let json;
+        try {
+            json = JSON.parse(buffered);
+        } catch (err) {
+            return cb(
+                boom.wrap(new Error('Payload provided to /feed was empty'), 400)
+            );
+        }
+        if (json.length) {
+            for (const chunk of this.chunks) {
+                this.push(chunk);
+            }
+            cb();
+        } else {
+            cb(
+                boom.wrap(new Error('Payload provided to /feed was empty'), 400)
+            );
+        }
+    }
+}
```

Comment:
This lovely piece of funk is obscene for what it does. Namely ensuring that the payload in the stream is not empty and bail if it is. Please someone for the love of all thats good tell me I'm a retard and theres a better way to do this.